### ISSUE TITLE
Nodes and layers

### ISSUE THREAD

**Issue by torzdf** [deepfakes/faceswap-playground#78]
Being a bit of a newb when it comes to machine learning, I'd like to play around with the nodes and layers a bit, but I'm not sure what to change.

I assume that nodes is ENCODER_DIM, but where are layers set?

If I were to bump ENCODER_DIM to 2048, I assume that an extra line would need to be added to Encoder like so:
```
x = self.conv(1024)(x)
x = self.conv(2048)(x)                              # << This line added to increase nodes
x = Dense(ENCODER_DIM)(Flatten()(x))
```

Would this be correct?

Cheers

**Comment by Clorr** [comment]
It is correct as it will run. However you should ask about the effect it has. Adding one more layer may give you more features but has 2 drawbacks:
- it takes more memory 
- it gives more space for embeddings and you may end up with an overfitted model

**Comment by torzdf** [comment]
Thanks for the info. I appreciate that. It's more a learning exercise for me than improving results. I'm just curious to see how the nodes and layers effect training, and this seems a good place to do it.

So would I be correct in assuming that the low_mem version has 3 layers and the original has 4. 

If, for example (and I'm not saying this would be a good idea, just making sure I have grasped the concept correctly), I wanted to train at 1024 nodes, but with 3 layers, would it be correct to amend as:

`ENCODER_DIM = 1024`
...
```
    def Encoder(self):
        input_ = Input(shape=IMAGE_SHAPE)
        x = input_
#        x = self.conv(128)(x)                              # <<< Remove first layer
        x = self.conv(256)(x)
        x = self.conv(512)(x)
        x = self.conv(1024)(x)
        x = Dense(ENCODER_DIM)(Flatten()(x))
        x = Dense(4 * 4 * 1024)(x)
        x = Reshape((4, 4, 1024))(x)
        x = self.upscale(512)(x)
return KerasModel(input_, x)
```

**Comment by Clorr** [comment]
As I said, I can't say if it is correct or not, it is just different and will behave differently but I can't tell you what will happen just by looking.

You can freely play with the model, the only constraint is that the layers shapes are compatible. Use `.summary()` to draw layers and see the final shape of your model.

Once you have created a new model, just run it and see how it behaves. The two worst things that can happen are:
- Loss values never go down, which means your model can't learn good features
- Model get's overfit and works very well on your training set but is bad in real life conditions
